Classes/Domain/Repository/BookmarkRepository.php

```diff
@@ -78,13 +78,17 @@ public function removeBookmark($document, $feUserUid)
 
     /**
      * @param int $feUserUid
-     * @param Document $document
+     * @param Document|string $document
      * @return bool
      * @throws \TYPO3\CMS\Extbase\Persistence\Exception\IllegalObjectTypeException
      */
-    public function addBookmark($feUserUid, Document $document)
+    public function addBookmark($feUserUid, $document)
```

Comment:
A bookmark is nothing more than a document id and feuser id. There is a bookmark model but using it outside of the repository lead to duplicate code and this was a solution to avoid this.

Reply:
Sorry, I missed that there is a Domain/Model/Bookmark. So what is being duplicated? The Bookmark construction?

Reply:
The code to handle creating and removing a bookmark.

Reply:
As discussed this morning I changed the method signature, so addBookmark() and removeBookmark() are equal now. I also added some comments to explain the different document identifiers.